models/optional_modules/conversions/scratch/snowplow_unified_conversions_this_run.sql

```diff
@@ -0,0 +1,47 @@
+{#
+Copyright (c) 2023-present Snowplow Analytics Ltd. All rights reserved.
+This program is licensed to you under the Snowplow Personal and Academic License Version 1.0,
+and you may not use this file except in compliance with the Snowplow Personal and Academic License Version 1.0.
+You may obtain a copy of the Snowplow Personal and Academic License Version 1.0 at https://docs.snowplow.io/personal-and-academic-license-1.0/
+#}
+
+{{
+  config(
+    tags=["this_run"],
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with prep as (
+
+{%- for conv_def in var('snowplow__conversion_events', []) %}
+
+  select
+    
+    session_identifier,
+    user_identifier,
+    {% if var('snowplow__conversion_stitching') %}
+    -- updated with mapping as part of post hook on derived conversions table
+    cast(user_identifier as {{ snowplow_utils.type_max_string() }}) as stitched_user_id,
+    {% endif %}
+    cv_{{ conv_def['name'] }}_volume as cv_volume,
+    cv_{{ conv_def['name'] }}_values as cv_values,
+    cv_{{ conv_def['name'] }}_total as cv_total,
+    cv_{{ conv_def['name'] }}_events as cv_events,
+    cv_{{ conv_def['name'] }}_first_conversion as cv_first_conversion,
+    cv_{{ conv_def['name'] }}_converted as cv_converted,
```

Comment:
Slight problem is not all conversions have all of these fields - in particular events I think is optional?

Reply:
good point, I will make these condtional